### PR TITLE
Infrastructure for Kafka and inter-bounded context communication

### PR DESCRIPTION
**Pull request by einari**
Putting in this P.R. to get verification that nothing is broken before I move on and actually hook this up and use it. 

The Kafka infrastructure is now separate and will be used by the text messaging / SMS infrastructure. 

While all B.C. related things is pulled out into its own infrastructure.

I've gone through and ran all the different core bounded contexts and it seems to be good for now.

### CODE REVIEW

**Comment by einari** [comment]
LOl.. Thanks for the thoroughness 👍

File: Source/Infrastructure/Kafka/KafkaModule.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+namespace Kafka
+{
+    public class KafkaModule : Autofac.Module
+    {
+        const string KAFKA_CONNECTIONSTRING = "KAFKA_CONNECTIONSTRING";
+
+        protected override void Load(ContainerBuilder builder)
+        {
+            var environmentVariables = Environment.GetEnvironmentVariables();
+            var kafkaConnectionString = "52.178.92.69:9092";
```
Comment (karolikl) [inline Source/Infrastructure/Kafka/KafkaModule.cs]: Even though this is the fallback used if the connectionsting is not passed in as an environment variable, should the default be hardcoded?

File: Source/Infrastructure/deploy/azure-secret.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: azure-storage-secret
+type: Opaque
+data:
+  azurestorageaccountname: Y2JzMmRldg==
+  azurestorageaccountkey: UmtpcUhsa3dsSnJhVUF4MW16OVBHdUoyUHQzejVEekU5eUk4ZXFTUVUzQVVzeTF6ZHBIbk0xN25NWGZ4OE50RGhwcVJSMWU2Mk91K0VVbFdMWjZLbmc9PQ==
```
Comment (karolikl) [inline Source/Infrastructure/deploy/azure-secret.yml]: I'm a little concerned about the security here, could we store it in Azure KeyVault or similar and retrieve the key from there?

File: Source/UserManagement/UserManagement.code-workspace [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+	"folders": [
+		{
+			"path": "/Users/einari/Projects/RedCross/cbs/Source/Infrastructure"
```
Comment (karolikl) [inline Source/UserManagement/UserManagement.code-workspace]: Not sure what this does, but the path is specific to your machine, can this be made generic?

File: Source/VolunteerReporting/VolunteerReporting.code-workspace [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+	"folders": [
+		{
+			"path": "."
+		},
+		{
+			"path": "/Users/einari/Projects/RedCross/cbs/Source/Infrastructure"
```
Comment (karolikl) [inline Source/VolunteerReporting/VolunteerReporting.code-workspace]: Not sure what this does, but the path is specific to your machine, can this be made generic?

File: Source/UserManagement/UserManagement.code-workspace (hunk 3 above)
Comment (einari) [reply Source/UserManagement/UserManagement.code-workspace]: Weird that VSCode puts in absolute paths - I've fixed these in an upcoming PR to be relative.

File: Source/Infrastructure/Kafka/KafkaModule.cs (hunk 1 above)
Comment (einari) [reply Source/Infrastructure/Kafka/KafkaModule.cs]: Agree, we should be using the .NET configuration where we can allow AppSettings and environment variables overrides - putting in an issue for it; #483

File: Source/Infrastructure/deploy/azure-secret.yml (hunk 2 above)
Comment (einari) [reply Source/Infrastructure/deploy/azure-secret.yml]: Yup.. This is very temporary - and absolutely madness of me to do. :).  I will delete the entire storage account in a couple of days - its sitting in a very different subscription. 
We should at the very minimum be using Kubernetes secrets for this and keep them outside the repo.

File: Source/VolunteerReporting/VolunteerReporting.code-workspace (hunk 4 above)
Comment (einari) [reply Source/VolunteerReporting/VolunteerReporting.code-workspace]: Workspaces are brilliant in VSCode - its VSCodes answer to SLN files - just slightly more readable and more oriented around folders.  But yeah, relative paths - coming up..

**Comment by Velcrow81** [comment]
No, the build fails after I have given my seal of approval. Utter horror :) Could you have a look @einari ? Have you forgotten to include stuff in solution files? 

"/workspace/Source/Admin/Admin.sln" (Build target) (1) ->
(Build target) -> 
  /workspace/Source/Admin/Admin.sln.metaproj : error MSB3202: The project file "/workspace/Source/Admin/../Infrastructure/Logging/Logging.csproj" was not found. [/workspace/Source/Admin/Admin.sln]
  /workspace/Source/Admin/Admin.sln.metaproj : error MSB3202: The project file "/workspace/Source/Admin/../Infrastructure/Kafka/Kafka.csproj" was not found. [/workspace/Source/Admin/Admin.sln]
  /workspace/Source/Admin/Admin.sln.metaproj : error MSB3202: The project file "/workspace/Source/Admin/../Infrastructure/Kafka.BoundedContexts/Kafka.BoundedContexts.csproj" was not found. [/workspace/Source/Admin/Admin.sln]

**Pull request merged**